### DISCUSSION TITLE
How can I run/debug an app on an Android phone?

### DISCUSSION THREAD

**Discussion by ygoe** [unoplatform/uno#18993] [Q&A]
### Your feedback

I tried to find out how to run an app from Visual Studio on my Android phone. I only found out how to publish it as .apk file to install. But it tkaes very long and copying the file over is an extra step. I'm looking for a debug process where I can immediately start my app on the phone, make changes and repeat. Is that possible? I think it was possible with MAUI at some point but I never really used that (or any Android development).

I have enabled Wifi debugging on the phone, so it should be available somehow. Don't want to connect a USB cable as that port is normally covered.

**Comment by ygoe** [comment]
PS: I forgot to mention the Android app installation process that includes some lengthy scanning. And now I see that my app doesn't work. It crashes in my first button event handler. So I really need a debugger to get it working.

**Comment by ygoe** [comment]
I found [instructions for MAUI](https://learn.microsoft.com/en-us/dotnet/maui/android/device/setup?view=net-maui-8.0#enable-wifi-debugging) and followed them, it's working for now. Maybe it could be added to the Uno documentation to make it easier.

**Comment by ajpinedam** [comment]
Hi @ygoe 

To run an app on an Android device is the same process you would follow for a simulator.

Your phone needs to be in developer mode
connect your phone to your Windows computer and it will show on the available devices on the Visual Studio Run as it it shown in the Uno Documentation; https://platform.uno/docs/articles/create-an-app-vs2022.html?tabs=Android#debug-the-app

Let me know if you find any more issues.

**Comment by ajpinedam** [comment]
I followed the steps above: The "OnePlus GM1917" is my personal Android device showing on the list of available devices.

<img width="513" alt="image" src="https://github.com/user-attachments/assets/509dd195-5de9-4486-9bc4-d23f725ae9b6">

**Comment by ygoe** [comment]
Hm, yes, that works, too. My app is about wireless communication and a presence sensor, so testing it involves a lot of walking around. Thus I need the WiFi option.
